standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```
@@ -1802,57 +1808,73 @@ private long partsFoundForPartitions(
-        // count(\"PARTITION_NAME\")==partNames.size()
-        // Or, extrapolation is not possible for this column if
-        // count(\"PARTITION_NAME\")<2
-        Long count = MetastoreDirectSqlUtils.extractSqlLong(row[2]);
-        if (count == partNames.size() || count < 2) {
-          noExtraColumnNames.add(colName);
-        } else {
-          extraColumnNameTypeParts.put(colName, new String[] { colType, String.valueOf(count) });
+      try(Query query = pm.newQuery("javax.jdo.query.SQL", queryText);) {
+        qResult = executeWithArray(query,
+            prepareParams(catName, dbName, tableName, partNames, colNames,
+                engine), queryText);
+        end = doTrace ? System.nanoTime() : 0;
+        MetastoreDirectSqlUtils.timingTrace(doTrace, queryText, start, end);
+        if (qResult == null) {
+          query.closeAll();
+          return Collections.emptyList();
+        }
+
+        List<Object[]> list = MetastoreDirectSqlUtils.ensureList(qResult);
+        for (Object[] row : list) {
+          String colName = (String) row[0];
+          String colType = (String) row[1];
+          // Extrapolation is not needed for this column if
+          // count(\"PARTITION_NAME\")==partNames.size()
+          // Or, extrapolation is not possible for this column if
+          // count(\"PARTITION_NAME\")<2
+          Long count = MetastoreDirectSqlUtils.extractSqlLong(row[2]);
+          if (count == partNames.size() || count < 2) {
+            noExtraColumnNames.add(colName);
+          } else {
+            extraColumnNameTypeParts
+                .put(colName, new String[] {colType, String.valueOf(count)});
+          }
+          Deadline.checkTimeout();
+        }
+      } catch (Exception e) {
+        if (e instanceof MetaException) {
+          throw (MetaException) e;
         }
-        Deadline.checkTimeout();
       }
-      query.closeAll();
       // Extrapolation is not needed for columns noExtraColumnNames
```

swallows exceptions which are not metaexceptions